apps/api/src/app/workflows-v2/dto/workflow.dto.ts

```diff
@@ -0,0 +1,140 @@
+import { IsArray, IsBoolean, IsDefined, IsEnum, IsOptional, IsString, ValidateNested } from 'class-validator';
+import { Type } from 'class-transformer';
+
+import { JsonSchema } from '@novu/framework';
+import { ChannelTypeEnum, StepTypeEnum, WorkflowOriginEnum } from '@novu/shared';
+
+class Schema {
+  schema: JsonSchema;
+}
+export type SingleChannelPreference = {
+  defaultValue: boolean;
+  readOnly: boolean;
+};
+
+export type WorkflowPreferencesDto = {
+  defaultWorkflowPreference: SingleChannelPreference;
+  channels: {
+    [key in (typeof ChannelTypeEnum)[keyof typeof ChannelTypeEnum]]: SingleChannelPreference;
+  };
+};
+
+export type CreateWorkflowDto = Omit<WorkflowDto, '_id'>;
+
+export type UpdateWorkflowDto = Omit<WorkflowDto, '_id'>;
+
+export type WorkflowResponseDto = WorkflowDto & {
+  updatedAt: string;
+};
+
+export type ListWorkflowResponse = {
+  workflowSummaries: MinifiedResponseWorkflowDto[];
+  totalResults: number;
+};
+
+export type MinifiedResponseWorkflowDto = Pick<WorkflowResponseDto, 'workflowId' | 'tags' | 'updatedAt' | '_id'> & {
+  stepSummery: StepTypeEnum[];
+};
+
+export class StepDto {
+  @IsString()
+  @IsDefined()
+  code: string;
+
+  @IsString()
+  @IsDefined()
+  stepId: string;
+
+  @IsString()
+  @IsDefined()
+  type: StepTypeEnum;
+
+  @IsOptional()
+  @ValidateNested()
+  @Type(() => Schema)
+  controls?: Schema;
+
+  @IsOptional()
+  @ValidateNested()
+  @Type(() => ReplyCallbackDto)
+  replyCallback?: ReplyCallbackDto;
+
+  @IsOptional()
+  @IsBoolean()
+  active?: boolean;
+
+  @IsOptional()
+  @IsBoolean()
+  shouldStopOnFail?: boolean;
+}
+
+export class WorkflowDto {
+  @IsString()
+  @IsDefined()
+  _id: string;
+
+  @IsString()
+  @IsDefined()
+  workflowId: string;
+
+  @IsOptional()
+  @ValidateNested()
+  @Type(() => Schema)
+  payload?: Schema;
+
+  @IsOptional()
+  @ValidateNested()
+  @Type(() => Schema)
+  controls?: Schema;
+
+  @IsOptional()
+  @IsArray()
+  @IsString({ each: true })
+  tags?: string[];
+
+  @IsOptional()
+  @IsBoolean()
+  critical?: boolean;
+
+  @IsOptional()
+  @IsBoolean()
+  active?: boolean;
+
+  @IsDefined()
+  @IsString()
+  notificationGroupId: string;
```

Comment:
I think we can also remove this property. Can you share the purpose of this property for V1 Workflows?

Reply:
The idea was to have a group functionality in the UI, and I'm fairly certain there were a few use cases for it. However, since it was a while ago, I'm having trouble remembering those use cases.